helper.go

```diff
@@ -463,41 +463,68 @@ func marshalObjectIdentifier(oid []int) (ret []byte, err error) {
 	return
 }
 
-func marshalOID(oid string) ([]byte, error) {
+// oidToIntArray is a helper function for marshalOID and takes an OID string
+// converts it to an array of integers and returns an int slice or an error
+func oidToIntArray(oid string, sep string) ([]int, error) {
 	var err error
-
-	// Encode the oid
-	oid = strings.Trim(oid, ".")
-	oidParts := strings.Split(oid, ".")
-	oidBytes := make([]int, len(oidParts))
-
-	// Convert the string OID to an array of integers
-	for i := 0; i < len(oidParts); i++ {
-		oidBytes[i], err = strconv.Atoi(oidParts[i])
-		if err != nil {
-			return nil, fmt.Errorf("unable to parse OID: %w", err)
+	n := strings.Count(oid, sep) + 1
+	a := make([]int, n)
+	n--
+	i := 0
+	for i < n {
+		m := strings.Index(oid, sep)
+		if m < 0 {
+			break
 		}
+		if a[i], err = strconv.Atoi(oid[:m]); err != nil {
+			return nil, fmt.Errorf("value out of range")
+		}
+		if uint32(a[i]) < MinObjectSubIdentifierValue || uint32(a[i]) > MaxObjectSubIdentifierValue {
+			return nil, fmt.Errorf("value out of range")
+		}
+		oid = oid[m+len(sep):]
+		i++
+	}
+	if a[i], err = strconv.Atoi(oid); err != nil {
+		return nil, fmt.Errorf("value out of range")
 	}
+	if uint32(a[i]) < MinObjectSubIdentifierValue || uint32(a[i]) > MaxObjectSubIdentifierValue {
+		return nil, fmt.Errorf("value out of range")
+	}
+	return a[:i+1], nil
+}
 
-	mOid, err := marshalObjectIdentifier(oidBytes)
+func marshalOID(oid string) ([]byte, error) {
```

Comment:
I don't see how this is doing any marshalling, all this function effectively does is strip a leading period from what is already an oid.

Reply:
I have not changed the names of the functions. I just changed part of the code from this function and moved it into a new function oidToIntArray () for readability.

Reply:
That doesn't really address my point, if you care about efficiency you can make this function a lot simpler by deleting most of its code. The name can also be improved.

Reply:
@brian-brazil Can you be more explicit about what you mean by "deleting most of the code"?

Reply:
Per above "all this function effectively does is strip a leading period from what is already an oid."